src/StrawberryfieldUtilityService.php

```diff
@@ -346,54 +367,48 @@ public function getCountByProcessorInSolr(EntityInterface $entity, string $proce
-      return FALSE;
-    }
-
-    // TODO is there an easier way to determine if the file path is valid and writable?
-    //   \Drupal::service('path.validator')->isValid($path) always returns FALSE,
-    //   and \Drupal::service('stream_wrapper_manager')->isValidUri($path) always returns TRUE!
-
-    $test_path = $scheme . "://" . $path;
 
-    // FileSystemInterface::prepareDirectory with the MODIFY_PERMISSIONS flag checks to see if it's present and able to be written to.
-    $valid = \Drupal::service('file_system')->prepareDirectory($test_path, FileSystemInterface::MODIFY_PERMISSIONS);
-    if(!$valid) {
-      // If not present or able to be written to, let's check to see if we can create it.
-      $valid = \Drupal::service('file_system')->prepareDirectory($test_path, FileSystemInterface::CREATE_DIRECTORY);
-      if($valid) {
-        // Successfully created the directory, so we delete it now.
-        \Drupal::service('file_system')->rmdir($test_path);
-      }
-    }
-    return !empty($valid);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function s3FilePathIsValid(string $path): bool {
-    // Empty path is always valid.
-    if(empty($path)) {
+    $known_schemes = array_merge(['s3'], array_keys(\Drupal::service('stream_wrapper_manager')->getWrappers(StreamWrapperInterface::LOCAL)));
+    if(!in_array($scheme, $known_schemes)) {
+      // Can't flag as invalid if we don't handle this scheme, so just return TRUE.
       return TRUE;
     }
 
+    // S3 path length limit is 63 characters.
+    if(strlen($path) > 63) {
+      return FALSE;
+    }
     // Split the path into virtual directory names and check each one. If
     // any are not valid, exit the loop and return FALSE.
     $path = explode("/", $path);
-    foreach($path as $path_part) {
-      $valid = preg_match(self::S3PATHPATTERN, $path_part, $matches);
-      if(!$valid) {
+    foreach ($path as $path_part) {
+      $valid = preg_match(self::VALID_S3_PREFIX_PATTERN, $path_part, $matches);
+      if (!$valid) {
         return FALSE;
       }
     }
-    return TRUE;
+
+    // The path validates against our test pattern, but we don't know if it can
+    // actually be written to by the file system. We'll check for that here.
+    // TODO: Is there a non-invasive way to test this?
+    $test_path = $scheme . "://" . $path;
+    // FileSystemInterface::prepareDirectory with the MODIFY_PERMISSIONS flag checks
+    // to see if it's present and able to be written to.
+    $valid = $this->fileSystem->prepareDirectory($test_path, FileSystemInterface::MODIFY_PERMISSIONS);
+    if (!$valid) {
+      // If not present or able to be written to, let's check to see if we can create it.
+      $valid = $this->fileSystem->prepareDirectory($test_path, FileSystemInterface::CREATE_DIRECTORY);
+      if ($valid) {
+        // Successfully created the directory, so we delete it now.
+        $this->fileSystem->rmdir($test_path);
+      }
+    }
+    return !empty($valid);
```

Comment:
@patdunlavey better (for readability) to set $valid to FALSE at the beginning. That way you always return$valid